client/landing/stepper/declarative-flow/internals/index.tsx

```diff
@@ -59,6 +59,11 @@ export const FlowRenderer: React.FC< { flow: Flow } > = ( { flow } ) => {
 	);
 	const previousProgressValue = stepProgress ? previousProgress / stepProgress.count : 0;
 
+	// this pre-loads all the lazy steps down the flow.
+	useEffect( () => {
+		Promise.all( flowSteps.map( ( step ) => 'asyncComponent' in step && step.asyncComponent() ) );
+	}, stepPaths );
```

Comment:
`stepPaths` is an array of steps slugs, they're strings (primitives). This ensures the effect doesn't run unless the steps actually change.